### ISSUE TITLE
No encryption

### ISSUE THREAD

**Issue by BadKaiPanda** [onionApps/Chat.onion#3]
'Anonymous and fully encrypted '
Have you taken out the encryption and updated the app on google as I downloaded the source and ran it and had the google app on the phone and they both talked fine, I looked at your code and there is no encryption at all.

There is no encryption in any of these files what I can see of, you have encoding but no encryption what even if the messages was encrypted there subject to normal data base local hacks

Also can you add a option what is off record chat, so if the user if online it pols the messages to the screen but not in a DB or cashe at all, so means we can close that chat and its gone, nice clean.

Also you don't state what is the recommended API for this build.

**Comment by sijanec** [comment]
As far as I know, you are wrong. There is encryption because everything in Tor is encrypted, your ID (onion domain) is your public key, private key is in your storage

**Comment by BadKaiPanda** [comment]
> As far as I know, you are wrong. There is encryption because everything in Tor is encrypted, your ID (onion domain) is your public key, private key is in your storage

Please read the code and try that one again there is no lib for encryption or using public or private keys at all and I can tell you know I already created tests you can create a basic web ui to capture the chat there no NO encryption, please read and learn what your using before you comment, I dropped my edited version for the simple fact where I live they are thinking about just blocking tor traffic so my aims became to find away to chat under there radar without them know without the need of tor. 

Edit also no tor connection is 100% safe as you don't know who owns your exit reply, and tor packages can be sniffed we are protected with a basic layer of protection with tor nothing more nothing less don't ever believe your safe on tor, deff on on a android device what logs all your actions all the way down to how hard you press the screen and YES normal android devices do store this and do send this to google and other 3ed parties.

**Comment by marataziat** [comment]
No dude, there is no encryption. Its like to say I use https:// so my app is encrypted. Dude its not. You should add some gpg encryton!

**Comment by CrazyBoyFeng** [comment]
> there is no lib for encryption or using public or private keys at all 

There is no encryption in the code, but when you send data via the tor network, tor will encrypt the data. The relay node cannot decrypt the data.

> Edit also no tor connection is 100% safe as you don't know who owns your exit reply.

When you communicate with the tor onion service, the exit node is not used. The tor onion service is a node inside the tor network.  
As for the security of android, that is not something that tor and this project can solve. If there is spyware in the system, it is not safe to use any software.

**Comment by sijanec** [comment]
> Please read the code and try that one again there is no lib for encryption or using public or private keys at all and I can tell you know I already created tests you can create a basic web ui to capture the chat there no NO encryption, please read and learn what your using before you comment, I dropped my edited version for the simple fact where I live they are thinking about just blocking tor traffic so my aims became to find away to chat under there radar without them know without the need of tor.

Please do not insult others and do not insist on your idea that Tor
connections are not encrypted. For additional information I advise you
to check https://en.wikipedia.org/wiki/Onion_routing and
https://en.wikipedia.org/wiki/Tor_(anonymity_network). Public and
private keys exist in form of public and private keys of the Tor
address. The "lib" you were looking for in the source code is the Tor
library. If you live in a country where Tor traffic is blocked,
Chat.onion will not work as it uses the Tor protocol. But only
blocking is possible; there is no way for someone to read your
messages unless they manage to find a way or a backdoor in the Tor
protocol or it's upstream libraries.

> Its like to say I use https:// so my app is encrypted.

The HTTPS protocol implies being sent over a secure channel, which is
in most cases SSL or TLS. SSL and TLS transfer encrypted data over the
wire so messages to your "App" are encrypted.

> You should add some gpg encryton!

This would just add unnecessary connection and processing overhead. If
you want to verify that someone whose GPG key you know is talking to
you via Chat.onion, ask them to sign a GPG message confirming their
onion address; after that you can use plain Tor to communicate without
the need for GPG.


V V pon., 24. maj 2021 ob 11:18 je oseba 狂男风
***@***.***> napisala:

**Comment by CrazyBoyFeng** [comment]
To @sijanec:
The notification email you received has no format, so you maybe misunderstood me. If you open the webpage, you can see that I am not insulting you, nor insist that Tor is not encrypted. I am "insulting" @BadKaiPanda, the author of this issue, and insist that Tor is encrypted.

**Comment by sijanec** [comment]
Yeah, sorry, you're right, it's not visible to me who is being quoted in
the email digest.

**Comment by RegressiaUK** [comment]
No wonder this project never got updated more if users like you guys are using it, just because the connection is encrypted it doesn't mean your date is safe, you have to trust your system and the end system and the app to all respect that.

There are many real big tor network message tools ALL via .onion only traffic and ALL of them use end to end to end encryption on top of the .onion encryption.

The app does not include this the OP is correct with this statement, and tor isn't 100% 

So Yes tor .onion offers a layer of encryption but this is not what the OP said they said the software it's self does not and this is also true.

Also why are you insulting anyone, the OP didn't even post to you two it was to the creator about missing options, you was fucking rude and wrong in your replies because the OP was correct the tool as no encryption, replying on just tor to keep your data safe is a foolish. 

I know this is a old topic but I was shocked about these posts and tor is not 100% tor make this clear on there site, tor is not all green and was originally made by governments and the army and is still funded by them to this day.

So having your own layer of protection is important.
